### ISSUE TITLE
scheduled scrub failure re non-string (type list)

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#1798]
Thanks to forum members Dragon2611, magicalyak, and peter in the following forum thread for highlighting this issue. Currently this has only been confirmed by magicalyak and peter as Dragon2611 may have been affected by a now fixed issue relating to pool API changes.

This issue relates to an as yet unidentified circumstance where the pid (pool id) passed to:
PoolScrubView._validate_pool() in src/rockstor/storageadmin/views/pool_scrub.py
is actually a list rather than the expected unicode of the pool database id.

The following was reported by magicalyak and peter in the referenced thread:
```
TypeError: __str__ returned non-string (type list)
Logged from file pool_scrub.py, line 78
Traceback (most recent call last):
  File "/usr/lib64/python2.7/logging/handlers.py", line 76, in emit
    if self.shouldRollover(record):
  File "/usr/lib64/python2.7/logging/handlers.py", line 154, in shouldRollover
    msg = "%s\n" % self.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 724, in format
    return fmt.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 464, in format
    record.message = record.getMessage()
  File "/usr/lib64/python2.7/logging/__init__.py", line 324, in getMessage
    msg = str(self.msg)
TypeError: __str__ returned non-string (type list)
Logged from file pool_scrub.py, line 39
```

Please update the following forum thread with this issues resolution:
https://forum.rockstor.com/t/scheduled-scrub-not-running/3608

**Comment by phillxnet** [comment]
My initial suspicion re the cause of this variable anomaly is a suspected inadvertent variable name clash.

I am currently looking into this issue but do not yet have a reproducer.

**Comment by phillxnet** [comment]
I am transferring my attention to the linked issue #1800 as I have a reproducer and this issue would best be addressed after the scheduled scrub improvements planed as part of that issue.

**Comment by phillxnet** [comment]
I have now managed to reproduce this issue, it appears (by my own miscommunication in the indicated forum thread) that the relevant rockstor release version was not installed (>= 3.9.1-2) and the existing scrub jobs that caused this issue (by using the old pool name based API) were recreated but not after the relevant fixes were in place.

Issue duplication was achieve by hard coding the url used by the scrub manager to include names not numbers and the issue email was duplicated along with identical rockstor log output to that provided by @magicalyak in the indicated forum thread which lead to the discovery of my prior miss-communication.

Closing as already fixed in (>= 3.9.1-2) which includes the fix provided by:

pr #1760 "fix scheduled scrub regression re pool api changes. Fixes #1759"

with the caveat that a re-creation of all existing scrub jobs there-after is required.

N.B. The next stable release announcement should indicate a requirement to remove and re-create all existing scrub tasks to inherit the pool API change regression re scheduled tasks fix.
